internal/campaigns/executor_test.go

```diff
@@ -116,6 +118,21 @@ func TestExecutor_Integration(t *testing.T) {
 				srcCLIRepo.ID: []string{"main.go", "modified-main.go.md", "added-modified-main.go.md"},
 			},
 		},
+		{
+			name:  "empty",
+			repos: []*graphql.Repository{srcCLIRepo},
+			archives: []mockRepoArchive{
+				{repo: srcCLIRepo, files: map[string]string{
+					"README.md": "# Welcome to the README\n",
+					"main.go":   "package main\n\nfunc main() {\n\tfmt.Println(     \"Hello World\")\n}\n",
+				}},
+			},
+			steps: []Step{
+				{Run: `true`, Container: "doesntmatter:13"},
+			},
+			// No changesets should be generated.
+			wantFilesChanged: map[string][]string{},
```

Comment:
I don't 100% understand how this tests the behavior is right but I trust you that it does what we look for. Regardless this is a good test. 
I assume this would have failed before the fix because the hot run is going to yield a change? Even though it was empty?  Not sure why it would show any files changed when there was an empty diff.

Reply:
Each entry in `wantFilesChanged` represents a changeset spec. In this case, since `true` won't change anything, the test _should_ hit the empty diff code paths and not generate changeset specs, even though the cache is populated.

This does indeed fail on the old `main`:

```

~/trees/sourcegraph/src-cli on  main via 🐹 v1.15.5 ❯ go test -race ./...
ok  	github.com/sourcegraph/src-cli/cmd/src	0.224s
ok  	github.com/sourcegraph/src-cli/internal/api	(cached)
--- FAIL: TestExecutor_Integration (0.46s)
    --- FAIL: TestExecutor_Integration/empty (0.04s)
        executor_test.go:189: wrong number of changeset specs. want=0, have=1
        --- FAIL: TestExecutor_Integration/empty/warm_cache (0.00s)
            testing.go:1048: test executed panic(nil) or runtime.Goexit: subtest may have called FailNow on a parent test
FAIL
FAIL	github.com/sourcegraph/src-cli/internal/campaigns	0.565s
?   	github.com/sourcegraph/src-cli/internal/campaigns/graphql	[no test files]
ok  	github.com/sourcegraph/src-cli/internal/codeintel	(cached)
?   	github.com/sourcegraph/src-cli/internal/output	[no test files]
ok  	github.com/sourcegraph/src-cli/internal/servegit	(cached)
?   	github.com/sourcegraph/src-cli/schema	[no test files]
FAIL
```

Reply:
Ahh okay makes sense. Thanks for confirming!